ui/src/features/components/Report/compareReports.js

```diff
@@ -282,8 +320,55 @@ const mapDispatchToProps = {
     clearAggregateReportAndBenchmark: Actions.clearAggregateReportAndBenchmark,
 };
 
-const Block = ({header, dataList, style = {}}) => {
+const Symbol = ({element, onEditSymbol, index}) => {
+    const [editMode, setEditMode] = useState(false);
+    const [value, setValue] = useState(element);
+
+    const onKeyDown = (evt) => {
+        if (evt.key === 'Enter') {
+            save();
+        }
+    }
+
+    const save = () => {
+        if (editMode) {
+            setEditMode(false);
+            onEditSymbol(value, index);
+        }
+    }
+
+    return (
+        <div
+            style={{display: 'flex', flex: 1, alignItems: 'center', cursor: 'pointer'}}
+            onClick={() => setEditMode(true)}
+        >
+            {editMode
+            ? (
+                <ClickOutHandler onClickOut={save}>
+                    <TextArea
+                        value={value}
+                        style={{lineHeight: 'normal', width: '50px', margin: 'auto', maxHeight: '20px', paddingTop:'0px', paddingBottom: '0px', }}
+                        onKeyDown={onKeyDown}
+                        onChange={(evt, value) => setValue(evt.target.value)}
+                    />
+                </ClickOutHandler>
+            )
+            : value}
+        </div>
+    )
+}
+
+const Block = ({header, dataList, style = {}, onEditSymbol}) => {
     const headerStyle = {color: '#577DFE', fontWeight: '500'};
+    const [isSymbol, setIsSymbol] = useState(false);
```

Comment:
lets leave this component stateless by using isSymbol variable - like const isSymbol=header === 'Symbol' or just pass a prop isSymbol along with onEditSymbol

Reply:
Done